src/dbnode/storage/index/mutable_segments.go

```diff
@@ -234,7 +238,16 @@ func (m *mutableSegments) seriesActive(d doc.Metadata) bool {
 		return true
 	}
 
-	return !d.OnIndexSeries.TryMarkIndexGarbageCollected()
+	marked, reconciled := d.OnIndexSeries.TryMarkIndexGarbageCollected()
```

Comment:
are we just returning `reconciled` to increment metrics? can we just increment the metric inside the function instead? it seems to complicate the api (2 bools is quite error prone)

Reply:
I was trying to avoid creating new metrics on, or passing metrics into, the entry, since there are so many of them and that'd be more mem

Reply:
i don't follow how that creates more mem. a metric is just a reference

Reply:
Yeah so that'd be another reference per entry

Reply:
If you think that's not worth I'm fine going w/ adding a pointer to the metrics

Reply:
don't actually care that much. you could just pass the metrics to the method to increment

Reply:
Updated to pass the metrics 👍